src/softmax/softmax_rvv.c

```diff
@@ -0,0 +1,286 @@
+    const vfloat32m1_t poly_c_2 = __riscv_vfmv_v_f_f32m1(0x1.00001p-1, vlmax);
+    const vfloat32m1_t poly_c_3 = __riscv_vfmv_v_f_f32m1(0x1.55546ep-3, vlmax);
+    const vfloat32m1_t poly_c_4 = __riscv_vfmv_v_f_f32m1(0x1.554854p-5, vlmax);
+    const vfloat32m1_t poly_c_5 = __riscv_vfmv_v_f_f32m1(0x1.114662p-7, vlmax);
+    const vfloat32m1_t poly_c_6 = __riscv_vfmv_v_f_f32m1(0x1.7209d4p-10, vlmax);
+    const vfloat32m1_t poly_c_7 = __riscv_vfmv_v_f_f32m1(0x1.94480ap-13, vlmax);
+
+
+    size_t avl = n;
+    while (avl > 0) {
+        size_t vl = __riscv_vsetvl_e32m1(avl);
+        vfloat32m1_t vx = __riscv_vle32_v_f32m1(src, vl);
+        vx= __riscv_vfsub(vx, max_x, vl);
+
+        // argument reduction
+        vfloat32m1_t vxiln2 = __riscv_vfmul(vx, iln2, vl);
+        vint32m1_t       vk = __riscv_vfcvt_x_f_v_i32m1(vxiln2, vl); // require round to nearest mode
+        vfloat32m1_t    vfk = __riscv_vfcvt_f_x_v_f32m1(vk, vl);
+        // using vfnmsac.vf to evaluate r = x - k * log(2)
+        vfloat32m1_t     vr = __riscv_vfnmsac(vx, ln2, vfk, vl);
+
+        // polynomial approximation exp(r)
+        vfloat32m1_t poly_vr = poly_c_7;
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_6, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_5, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_4, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_3, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_2, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_1, vl);
+        poly_vr = __riscv_vfmadd(poly_vr, vr, poly_c_0, vl);
+
+        // reconstruction
+        const int exp_bias = 127;
+        vint32m1_t vbiased_exp = __riscv_vadd(vk, exp_bias, vl);
+        vint32m1_t vexp2_vk    = __riscv_vsll(vbiased_exp, 23, vl);
+        vfloat32m1_t vfexp2_vk;
+        vfexp2_vk = __riscv_vreinterpret_v_i32m1_f32m1(vexp2_vk);
+
+        vfloat32m1_t vexp_vx  = __riscv_vfmul(poly_vr, vfexp2_vk, vl);
+
+        // element-size reduction with redution accumulator
+        // tail-undisturbed is mandatory here to ensure that if vl is less
+        // than VLMAX then unaffacted sum terms are not changed.
+        vsum = __riscv_vfadd_vv_f32m1_tu(vsum, vsum, vexp_vx, vl);
+
+        __riscv_vse32(dst, vexp_vx, vl);
+        avl -= vl;
+        src += vl;
+        dst += vl;
+    }
+
+    vfloat32m1_t vredsum = __riscv_vfmv_v_f_f32m1(0.f, vlmax);
+    vredsum = __riscv_vfredusum_vs_f32m1_f32m1(vsum, vredsum, vlmax);
+
+    return __riscv_vfmv_f_s_f32m1_f32(vredsum);
+}
+
+
+/** implementation of softmax for binary32 with RVV-based normalization
+ * 
+ *  @param dst destination array
+ *  @param src source array
+ *  @param n   number of element(s)
+*/
+void softmax_rvv_norm_fp32(float* dst, float* src, size_t n)
+{
+    int i;
+
+    // computing the sum of exponentials
+    float sum = 0.f;
+    for (i = 0; i < n; ++i) {
+        dst[i] = quick_dirty_expf(src[i]); //expf(src[i]);
```

Comment:
Revert to version used in benchmarks:
```suggestion
        dst[i] = expf(src[i]);
```